File: my-folder/1036-rotting-oranges/solution.py
from collections import deque
class Solution:
    def orangesRotting(self, grid: List[List[int]]) -> int:
        rows=len(grid)
        cols=len(grid[0])
        fresh=0
        time=0
        queue=Deque()
        directions=[[0,1],[1,0],[-1,0],[0,-1]]
        for r in range(rows):
            for c in range(cols):
                if(grid[r][c]==1):
                    fresh+=1
                if(grid[r][c]==2):
                    queue.append((r,c))
        while queue and fresh>0:
            length=len(queue)
            for i in range(length):
                r,c=queue.popleft()
                for rd,rc in directions:
                    row=rd+r
                    col=rc+c
                    if (row in range(len(grid))
                        and col in range(len(grid[0]))
                        and grid[row][col] == 1
                    ):
                        grid[row][col] = 2
                        queue.append((row, col))
                        fresh -= 1
            time += 1
        return time if fresh == 0 else -1


                


                 
        return self.total_time
        

# grid = [2,1,1]
#        [1,1,0]
#        [0,1,1]
        

File: my-folder/1036-rotting-oranges/solution.java
class Solution {
    Queue<int[]> rottenOranges;
    public int orangesRotting(int[][] grid) {
        rottenOranges = new LinkedList<>();
        int freshOranges = 0;
        int totalTime = 0;
        for(int row = 0; row < grid.length; row++){
            for(int col = 0; col < grid[0].length; col++){
                if(grid[row][col] == 2){
                    rottenOranges.offer(new int[]{row, col});
                }
                else if(grid[row][col] == 1){
                    freshOranges += 1;
                }
            }
        }

        int[][] directions = {{-1, 0}, {1,0}, {0, 1}, {0, -1}};
        while(freshOranges > 0 && !rottenOranges.isEmpty()){
            int size = rottenOranges.size();
            for(int i = 0; i < size; i++){
                int[] curr = rottenOranges.poll();
                for(int[] direction : directions){
                    int row = curr[0] + direction[0];
                    int col = curr[1] + direction[1];
                    if(row >= 0 && col >= 0 && col < grid[0].length && row < grid.length){
                        if(grid[row][col] == 1){
                            freshOranges--;
                            grid[row][col] = 2;
                            rottenOranges.offer(new int[]{row, col});
                        }
                    }
                }  
            }
            totalTime++;
        }

        if(freshOranges != 0){
            return -1;
        }

        return totalTime;
    }
}
// total fresh oranges = 6
// what are rotten = 1



// [2,1,1]
// [1,1,0]
// [0,1,1]
